### DISCUSSION TITLE
How to set content-type header when streaming?

### DISCUSSION THREAD

**Discussion by PierBover** [honojs/hono#2770] [Q&A] [answered]
I'm trying to stream HTML but the browser is always receiving multiple content types.

> text/plain;charset=UTF-8, text/html; charset=UTF-8

I'm setting the content type on the context before returning the stream:

```
app.get('/hello', (context) => {
  context.res.headers.append('Content-Type', 'text/html; charset=UTF-8');
  
  return stream(context, async (stream) => {
    // send HTML string
  });
});
```

Using `streamText` just seems to override any content type I set. When using `stream` Hono seems to be appending mutliple content types.

**Comment by yusukebe** [comment, marked as answer]
HI @PierBover 

You can use `context.header()`:

```ts
context.header('Content-Type', 'text/html; charset=UTF-8')
```

**Reply by PierBover** [reply]
Thanks @yusukebe !

**Reply by yusukebe** [reply]
🎉
